### ISSUE TITLE
Support OCI labels

### ISSUE THREAD

**Issue by shibumi** [google/ko#538]
Hi,

I think it would be a cool addition to ko if it's possible to specify additional labels via command line flags or via the ko configuration file.
The OCI image-spec defines the following standard labels: https://github.com/opencontainers/image-spec/blob/main/annotations.md

**Comment by shibumi** [comment]
Closing this. I just realized there is an `--image-label` flag.

**Comment by imjasonh** [comment]
FWIW there's probably more we could do to detect and set these by default. Buildpacks is looking into a similar thing (https://github.com/buildpacks/rfcs/pull/196), and some of the stuff coming in Go 1.18 should make it easier to extract from the build environment. See #366 

Like with base image annotations (#221) we'd only be able to set the annotations on indexes if they're OCI indexes, which doesn't apply to the default distroless base image 😭. We should still be able to annotate images though, and we've toyed with the idea of always pushing OCI images (w/ annotations) even if the base was a non-OCI image (see #449), but this fails when pushing to some registries (e.g., quay.io), so it was reverted.
